src/gt4py/next/ffront/past_to_itir.py

```diff
@@ -169,10 +169,14 @@ def _gen_size_params_from_program(self, node: past.Program) -> list[itir.Sym]:
         """Generate symbols for each field param and dimension."""
         size_params = []
         for param in node.params:
-            if type_info.is_type_or_tuple_of_type(param.type, ts.FieldType):
-                fields_dims: list[list[common.Dimension]] = (
-                    type_info.primitive_constituents(param.type).getattr("dims").to_list()
-                )
+            fields_dims: list[list[common.Dimension]] = (
+                type_info.primitive_constituents(param.type)
+                .if_isinstance(ts.FieldType)
+                .getattr("dims")
+                .filter(lambda dims: len(dims) > 0)
+                .to_list()
+            )
+            if len(fields_dims) > 0:  # param has no field constituent otherwise
```

Comment:
```suggestion
            if len(fields_dims) > 0:  # otherwise `param` has no constituent which is of `FieldType`
```
I read it as constituent of a field